File: Sequence Valley/Queue/Q3. Implement Queue using Stacks/Q3_ImplementQueueusingStacks.py
class MyQueue(object):                     # Define Queue using two stacks
    def __init__(self):                    # Constructor
        self.in_stk = []                   # Stack used for adding elements
        self.out_stk = []                  # Stack used for removing elements

    # Push element x to the back of queue...
    def push(self, x):                     # Add an element to the queue
        self.in_stk.append(x)              # Push element into input stack

    # Remove the element from the front of the queue and returns it...
    def pop(self):                         # Remove front element
        self.peek()                        # Make sure output stack has elements
        return self.out_stk.pop()          # Pop and return front element

    # Get the front element...
    def peek(self):                        # View the front element
        if not self.out_stk:               # If output stack is empty
            while self.in_stk:             # Move all elements from input stack
                self.out_stk.append(self.in_stk.pop())  # Push into output stack
        return self.out_stk[-1]             # Return the front element

    # Return whether the queue is empty...
    def empty(self):                       # Check if queue is empty
        return not self.in_stk and not self.out_stk  # True if both stacks are empty

File: Sequence Valley/Queue/Q3. Implement Queue using Stacks/Q3_ImplementQueueusingStacks.java
import java.util.Stack;                     // Import Stack class

class MyQueue {                             // Define Queue using two stacks
    private Stack<Integer> in_stack;        // Stack for adding elements
    private Stack<Integer> out_stack;       // Stack for removing elements

    public MyQueue() {                      // Constructor
        in_stack = new Stack<>();           // Initialize input stack
        out_stack = new Stack<>();          // Initialize output stack
    }

    private void transfer() {               // Move elements between stacks
        while (!in_stack.isEmpty()) {       // While input stack has elements
            out_stack.push(in_stack.pop()); // Move element to output stack
        }
    }

    public void push(int x) {               // Add element to the queue
        in_stack.push(x);                   // Push element into input stack
    }

    public int pop() {                      // Remove front element
        if (out_stack.isEmpty()) {          // If output stack is empty
            transfer();                     // Move elements from input to output
        }
        return out_stack.pop();             // Remove and return front element
    }

    public int peek() {                     // Get front element
        if (out_stack.isEmpty()) {          // If output stack is empty
            transfer();                     // Move elements from input to output
        }
        return out_stack.peek();            // Return front element
    }

    public boolean empty() {                // Check if queue is empty
        return in_stack.isEmpty() && out_stack.isEmpty(); // True if both stacks are empty
    }
}
